### PR TITLE
Rod Cutting Problem using Dynamic Programming

### PR DESCRIPTION
**Pull request by michaelehab**
## Related Issue:
Rod Cutting Problem using Dynamic Programming #245 

## Proposed Changes
- Added the cpp solution file in /D-DynamicProgramming.
- Updated Readme.md file 

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the documentation.
- [x] 🎀 I have updated the documentation accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Function Screenshot
function_screenshot
## Output Screenshot
output_screenshot
## Readme Screenshot
readme_screenshot

### CODE REVIEW

**Comment by siddhi-244** [comment]
Resolve the conflicts

**Review by siddhi-244 — CHANGES_REQUESTED** [review]
Can uh explain the approach a bit more?

**Comment by michaelehab** [comment]
This is a famous dynamic programming problem, the approach is already explained in the readme file please check it carefully.

**Comment by siddhi-244** [comment]
> This is a famous dynamic programming problem, the approach is already explained in the readme file please check it carefully.

Do the changes and solve merge conflicts too !

**Comment by michaelehab** [comment]
@siddhi-244 
Excuse me, Which changes?
Alright, I'll solve the conflicts in the Readme file then you'll keep ignoring this PR for another 2 months.
I prefer that someone responsible checks this PR cause I don't think you're taking this seriously.
Thanks.
@Lakhankumawat

**Comment by siddhi-244** [comment]
> @siddhi-244 Excuse me, Which changes? Alright, I'll solve the conflicts in the Readme file then you'll keep ignoring this PR for another 2 months. I prefer that someone responsible checks this PR cause I don't think you're taking this seriously. Thanks. @Lakhankumawat

Sure !

File: D-DynamicProgramming/README.md [hunk 1]
```diff
@@ -1,3 +1,31 @@
...
+       
+#### Optimal Approach :  
+
+    We will solve this problem in a bottom-up manner. (iteratively)  
+    In the bottom-up approach, we solve smaller subproblems first, then move on to larger subproblems.  
+    The following bottom-up approach computes dp[i], which stores maximum profit achieved from the rod of length i from 1 to len. 
+    It uses the value of smaller values i already computed.
+    
+**Time Complexity :** O(n&sup2;)  
+**Space Complexity :** O(n)
```
Comment (Lakhankumawat) [inline D-DynamicProgramming/README.md:27]: 1. Decrease Time Complexity font size it looks unusually big for now

File: D-DynamicProgramming/README.md [hunk 2]
```diff
@@ -1,3 +1,31 @@
+
+## Rod Cutting 
+### Problem Statement
+Given a rod of length n and a list of rod prices of length i, where 1 <= i <= n, find the optimal way to cut the rod into smaller rods to maximize profit.
```
Comment (Lakhankumawat) [inline D-DynamicProgramming/README.md:4]: 2. Add some working image of rod cutting problem here like [this one](https://www.codesdope.com/staticroot/images/algorithm/rod1.png)

File: D-DynamicProgramming/README.md [hunk 3]
```diff
@@ -1,3 +1,31 @@
...
+#### Optimal Approach :  
+
+    We will solve this problem in a bottom-up manner. (iteratively)  
+    In the bottom-up approach, we solve smaller subproblems first, then move on to larger subproblems.  
+    The following bottom-up approach computes dp[i], which stores maximum profit achieved from the rod of length i from 1 to len. 
+    It uses the value of smaller values i already computed.
+    
+**Time Complexity :** O(n&sup2;)  
+**Space Complexity :** O(n)  
+=======
```
Comment (Lakhankumawat) [inline D-DynamicProgramming/README.md:28]: You need to follow particular guidelines for your README , please follow
these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) 
also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) 
for more please go through these [here](https://youtu.be/7UgC3HDETRA)

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]
Do the required changes , i'll resolve the merge conflicts for you and i'll merge your PR then

**Comment by michaelehab** [comment]
@Lakhankumawat Thank you for your clarification, Please check the latest commit and tell me if you need any further changes.

**Review by Lakhankumawat — APPROVED** [review]
LGTM ✨🎉

**Comment by Lakhankumawat** [comment]
@all-contributors please add @michaelehab code ,

**Pull request merged**